code/obj/railing.dm

```diff
@@ -0,0 +1,344 @@
+// this is such a dumb joke
+/obj/meme
+	name = "glorious"
+	desc = "FUCK WHAT THE JESUS IS THAT"
+	icon = 'icons/obj/160x160.dmi'
+	icon_state = "dumb-joke-left"
+	layer = 100
+	plane = 100
+	opacity = 0
+	density = 0
+	pixel_x = -64
+	pixel_y = -64
+
+	right
+		icon_state = "dumb-joke-right"
+
+/obj/railing
+	name = "railing"
+	desc = "Two sets of bars shooting onward with the sole goal of blocking you off. They can't stop you from vaulting over them though!"
+	anchored = 1
+	density = 1
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "railing"
+	layer = OBJ_LAYER
+	color = "#ffffff"
+	flags = FPRINT | USEDELAY | ON_BORDER | ALWAYS_SOLID_FLUID
+	event_handler_flags = USE_FLUID_ENTER | USE_CHECKEXIT | USE_CANPASS
+	dir = SOUTH
+	custom_suicide = 1
+	var/broken = 0
+
+	proc/layerify()
+		SPAWN_DBG(3 DECI SECONDS)
+		if (dir == SOUTH)
+			layer = MOB_LAYER + 0.1
+		else
+			layer = OBJ_LAYER
+
+	proc/railing_is_broken(obj/railing/The_Railing)
+		if(The_Railing.broken)
+			return 1
+		else
+			return 0
+
+	proc/railing_break(obj/railing/The_Railing)
+		if(!(railing_is_broken(The_Railing)))
+			The_Railing.broken = 1
+			The_Railing.density = 0
+			var/random_sprite = rand(1, 4)
+			The_Railing.icon_state = "railing-broken-" + "[random_sprite]"
+
+	proc/railing_fix(obj/railing/The_Railing)
+		if(railing_is_broken(The_Railing))
+			The_Railing.density = 1
+			The_Railing.broken = 0
```

Comment:
I think it'd make more sense to hae these not take an argument and instead use `src` instead of `The_Railing` unless I'm missing something?

Reply:
These procs are called from the action bars, and not the railings themselves. So I created a reference argument for the action bars to use, to prevent any strange accidents.

Reply:
`railing_fix` doesn't seem to be called from anywhere and the other two doesn't seem to be called from the action bars (only from ex_act, blob_act etc.). And even if they got called from action bars you'd probably be doing `the_railing.railing_fix(the_railing)` which is a bit redundant.